Modules/mmapmodule.c

```diff
@@ -255,6 +256,207 @@ do {                                                                    \
+            ULONG code = LsaNtStatusToWinError(status);                    \
+            PyErr_SetFromWindowsErr(code);                                 \
+        }                                                                  \
+        else if (record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION) {     \
+            PyErr_SetFromWindowsErr(ERROR_NOACCESS);                       \
+        }                                                                  \
+        return -1;                                                         \
+    }                                                                      \
+} while (0)
+#else
+#define HANDLE_INVALID_MEM(sourcecode)                                     \
+do {                                                                       \
+    sourcecode                                                             \
+} while (0)
+#endif
+
+#if defined(MS_WINDOWS) && !defined(DONT_USE_SEH)
+#define HANDLE_INVALID_MEM_METHOD(self, sourcecode)                                     \
+do {                                                                                    \
+    EXCEPTION_RECORD record;                                                            \
+    __try {                                                                             \
+        sourcecode                                                                      \
+    }                                                                                   \
+    __except (filter_page_exception_method(self, GetExceptionInformation(), &record)) { \
+        assert(record.ExceptionCode == EXCEPTION_IN_PAGE_ERROR ||                       \
+               record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION);                     \
+        if (record.ExceptionCode == EXCEPTION_IN_PAGE_ERROR) {                          \
+            NTSTATUS status = (NTSTATUS) record.ExceptionInformation[2];                \
+            ULONG code = LsaNtStatusToWinError(status);                                 \
+            PyErr_SetFromWindowsErr(code);                                              \
+        }                                                                               \
+        else if (record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION) {                  \
+            PyErr_SetFromWindowsErr(ERROR_NOACCESS);                                    \
+        }                                                                               \
+        return -1;                                                                      \
+    }                                                                                   \
+} while (0)
+#else
+#define HANDLE_INVALID_MEM_METHOD(self, sourcecode)                                     \
+do {                                                                                    \
+    sourcecode                                                                          \
```

Comment:
```suggestion
#define HANDLE_INVALID_MEM_METHOD(self, sourcecode)                           \
do {                                                                          \
    EXCEPTION_RECORD record;                                                  \
    __try {                                                                   \
        sourcecode                                                            \
    }                                                                         \
    __except (filter_page_exception_method(self, GetExceptionInformation(),   \
                                           &record)) {                        \
        assert(record.ExceptionCode == EXCEPTION_IN_PAGE_ERROR ||             \
               record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION);           \
        if (record.ExceptionCode == EXCEPTION_IN_PAGE_ERROR) {                \
            NTSTATUS status = (NTSTATUS) record.ExceptionInformation[2];      \
            ULONG code = LsaNtStatusToWinError(status);                       \
            PyErr_SetFromWindowsErr(code);                                    \
        }                                                                     \
        else if (record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION) {        \
            PyErr_SetFromWindowsErr(ERROR_NOACCESS);                          \
        }                                                                     \
        return -1;                                                            \
    }                                                                         \
} while (0)
#else
#define HANDLE_INVALID_MEM_METHOD(self, sourcecode)                           \
do {                                                                          \
    sourcecode                                                                \
```

I missed this earlier. It's just a superficial change to comply with the PEP 7 line-length limit of 79 characters.